keps/sig-node/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1067 @@
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `KubeletFineGrainedAuthz`
+  - Components depending on the feature gate: 
+    - `kubelet`
+    - `kube-apiserver`
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+While there will be no change to use-facing behavior as we will still send a 
+SubjectAccessReview with the `proxy` authorization subresource, we will be
+sending an extra SubjectAccessReview request.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes, but a workload that is only authorized to use the new authorization subresource
+will lose access and will need to update its RBAC Role to use `nodes/proxy`.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the kubelet feature-gate is re-enabled then kubelet will again start sending 
+2 SubjectAccessReview requests.
+
+If the kube-apiserver feature-gate is re-enabled then the ClusterRole `system:kubelet-api-admin`
+will be updated as described in the (Design Details section)[#design-details].
```

Comment:
Can you clarify here as well about what happens when the feature gate is enabled for only one of the components (kube-apiserver and kubelet)? Is there any recommended order of enabling/disabling?

Reply:
Good ideas! Done.